sql/core/src/test/scala/org/apache/spark/sql/streaming/CommitLogSuite.scala

```diff
@@ -108,6 +115,57 @@ class CommitLogSuite extends SparkFunSuite with SharedSparkSession {
     assert(commitMetadata.nextBatchWatermarkMs === 233)
     assert(commitMetadata.stateUniqueIds === Map.empty)
   }
+
+  // Test an old version of Spark can ser-de the new version of commit log,
+  // but running under V1 (i.e. no stateUniqueIds)
+  test("v1 Serde backward compatibility") {
+    // This is the json created by a V1 commit log
+    val commitLogV1WithStateUniqueId = """v1
+                        |{"nextBatchWatermarkMs":1,"stateUniqueIds":{}}""".stripMargin
```

Comment:
There should be another case where the commit log is generated by previous releases, likely looking lik:
```
"""v1
                         |{"nextBatchWatermarkMs":1}"""
```
Since we are already here we should cover that too.

Reply:
there is already such test